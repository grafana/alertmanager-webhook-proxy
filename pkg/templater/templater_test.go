package templater

import (
	"bytes"
	"encoding/json"
	"io/ioutil"
	"os"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/grafana/alertmanager-webhook-proxy/pkg/proxy"
)

func TestMain(t *testing.M) {
	v := t.Run()

	snaps.Clean()

	os.Exit(v)
}

func TestRender(t *testing.T) {
	tmpl, err := New("testdata/template.txt")

	if err != nil {
		t.Errorf("Error: %v", err)
	}

	payload, rErr := ioutil.ReadFile("testdata/sample.json")

	if rErr != nil {
		t.Error("failed to read sample payload")
	}

	reader := bytes.NewReader(payload)

	data, dErr := proxy.Decode(reader)

	if dErr != nil {
		t.Errorf("failed to decode: %v", dErr)
	}

	s, reErr := Render(tmpl, data)

	if reErr != nil {
		t.Errorf("failed to render: %v", reErr)
	}

	if !json.Valid([]byte(s)) {
		t.Errorf("invalid json: %v", s)
	}

	snaps.MatchSnapshot(t, s)
}
