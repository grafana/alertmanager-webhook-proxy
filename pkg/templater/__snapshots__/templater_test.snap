
[TestRender - 1]
{
  "alerts": [
    
    
    {
      "name": "Test",
      "severity": "<no value>",
      "description": "some description",
      "source": "grafana-cloud",
      "time": "2018-08-03 09:52:26.739266876 +0200 +0200",
      "extra": {
        "type": "application",
        "env": "<no value>",
        "ci": "<no value>"
      }
    }
    
    ,
    {
      "name": "Test",
      "severity": "<no value>",
      "description": "some description",
      "source": "grafana-cloud",
      "time": "2018-08-03 09:52:26.739266876 +0200 +0200",
      "extra": {
        "type": "application",
        "env": "<no value>",
        "ci": "<no value>"
      }
    }
    
    ,
    {
      "name": "Test",
      "severity": "<no value>",
      "description": "some description",
      "source": "grafana-cloud",
      "time": "2018-08-03 09:52:26.739266876 +0200 +0200",
      "extra": {
        "type": "application",
        "env": "<no value>",
        "ci": "<no value>"
      }
    }
    
  ]
}

---
